### PR TITLE
DevOps - Github Action to deploy & delete node-network

### PR DESCRIPTION
**Pull request by ahhda**
Adds a Github Action workflow dispatch to deploy the Pulumi stack for node-network #2406 

To test this Github Action

* Merge it to the default branch of the repository
* Set `PULUMI_ACCESS_TOKEN` in Repository Secrets
* To generate access token go to [https://app.pulumi.com/<USER>/settings/tokens](https://app.pulumi.com/<USER>/settings/tokens)
Screenshot 2021-08-26 at 6 22 46 PM

* Go to Actions > `Create node-network using Pulumi` workflow > Run workflow
* Enter your encryption key, this will be used to extract the contents of `chain-data.7z` file

### How to access the stack on your local

* Make sure you have Pulumi installed
* Set `PULUMI_ACCESS_TOKEN` environment variable in your terminal (`export PULUMI_ACCESS_TOKEN="pul..."`)
* Run `pulumi stack select <STACK_NAME>`
* To get output, run `pulumi stack output`
* To delete all the resources, run `pulumi destroy -s <STACK_NAME>`

### CODE REVIEW

File: .github/workflows/create-node-network-pulumi.yml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            echo "::add-mask::$ACCESS_TOKEN"
+            echo "PULUMI_ACCESS_TOKEN=$ACCESS_TOKEN" >> $GITHUB_ENV
+          fi
+
+      - name: Create local stack
+        run: |
+          npm install
+          pulumi login
+          pulumi stack select dev --non-interactive --create
+          pulumi config set-all --plaintext aws:region=us-east-1 \
```
Comment (mnaamani) [inline .github/workflows/create-node-network-pulumi.yml]: stack name "dev" should be a user input as well as other stack settings:
`numberOfValidators`, `networkSuffix`, and `nodeImage` with values you set here as defaults perhaps.

File: .github/workflows/create-node-network-pulumi.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+      - name: Set KUBECONFIG
+        run: |
+          pulumi stack output kubeconfig > kubeconfig.yml
+          echo "KUBECONFIG=$PWD/kubeconfig.yml" >> $GITHUB_ENV
+
+      - name: Get chain-data from the pod
+        run: |
+          kubectl config set-context --current --namespace=${{ steps.pulumi.outputs.namespaceName }}
+          kubectl cp $(kubectl get pods | grep rpc-node | awk '{print $1}'):/chain-data/chain-data.7z ./chain-data.7z
```
Comment (mnaamani) [inline .github/workflows/create-node-network-pulumi.yml:107]:
```suggestion
          kubectl cp $(kubectl get pods | grep rpc-node | awk '{print $1}'):chain-data/chain-data.7z ./chain-data.7z
```

Is this change needed?

File: .github/workflows/delete-node-network-pulumi.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+name: Delete Pulumi stack
+
+on:
+  workflow_dispatch:
+    inputs:
+      pulumiToken:
+        description: 'Pulumi Access Token'
+        required: false
+
```
Comment (mnaamani) [inline .github/workflows/delete-node-network-pulumi.yml:13]: stack name should be an input as well.

File: .github/workflows/create-node-network-pulumi.yml (hunk 1 above)
Comment (mnaamani) [inline .github/workflows/create-node-network-pulumi.yml]: Also the AWS region is configured in "Configure AWS credentials" step. Should we make sure a variable is used to ensure the same value is always used here as well?

**Review by mnaamani — CHANGES_REQUESTED** [review]

File: .github/workflows/create-node-network-pulumi.yml (hunk 2 above)
Comment (mnaamani) [reply .github/workflows/create-node-network-pulumi.yml:107]: Not required.

File: .github/workflows/delete-node-network-pulumi.yml (hunk 3 above)
Comment (ahhda) [reply .github/workflows/delete-node-network-pulumi.yml:13]: Done

File: .github/workflows/create-node-network-pulumi.yml (hunk 1 above)
Comment (ahhda) [reply .github/workflows/create-node-network-pulumi.yml]: Added all as inputs with default values

**Pull request closed without merging**